.github/workflows/pre-submit.actions.yml

```diff
@@ -115,6 +115,50 @@ jobs:
           SLSA_TOKEN: ${{ steps.setup.outputs.slsa-token }}
         run: ./.github/workflows/scripts/pre-submit.actions/setup-token.sh
 
+  verify-token:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      - id: token
+        run: |
+          set -euo pipefail
+          # Note: the token is always invalid due to the GitHub run verification.
+          token=$(cat .github/actions/verify-token/testdata/slsa-token)
+          verified_token=$(echo "$token" | cut -d '.' -f2 | base64 -d)
+          echo "unverified=$token" >> "$GITHUB_OUTPUT"
+          echo "verified=$verified_token" >> "$GITHUB_OUTPUT"
+      # Verify the values in the "verified" token.
+      - run: ./.github/workflows/scripts/schedule.actions/verify-token.sh
+        env:
+          VERIFIED_TOKEN: ${{ steps.token.outputs.verified }}
+          TOOL_REPOSITORY: "laurentsimon/slsa-delegated-tool"
+          TOOL_REF: "refs/heads/main"
+          TOOL_URI: "https://github.com/laurentsimon/slsa-delegated-tool/.github/workflows/tool1_slsa3.yml@refs/heads/main"
+      - id: verify
+        uses: ./.github/actions/verify-token
+        continue-on-error: true
+        with:
+          # Always fails due to the GitHub run mismatch.
+          slsa-unverified-token: ${{ steps.token.outputs.unverified }}
+          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+      - id: verify-mismatch-recipient
```

Comment:
Is this just going to fail anyway due to the run mismatch? are we actually testing what we think we're testing?

Reply:
could you maybe leverage a core.debug/core.setFailed somehow (maybe by running it as a node script instead)?

Reply:
I just copied from the schedule workflow. Agreed it's not very useful and maybe we should just rely on the scheduled runs... or do what Asra is suggesting. At this point maybe we should instead use unit test in the TS code itself?

Reply:
Yeah unit tests would be great if the code is modularized from signature checking to the policy.

I suggested a presubmit though  because this action was missing the action.yml! So long as *somewhere* we are running the action as an action in a test, we'd catch this. So maybe it's OK, since the actions will eventually be used in a workflow.

Reply:
mhhh, a pre-submit will always fail verification. I need to differentiate between errors, but I don't know if I can read the logs / error messages. Running as a node application does not require the action.yml, so we would not catch the problem.

Any other ideas?

Reply:
Are things like the run ID things we can fake by just setting the right GITHUB_* environment variables? I think that probably only solves one of the issues so won't get us all the way there anyway.

otherwise yeah, maybe unit tests are a better way to test the `verify-token` action.

Reply:
we cannot overwrite the `GITHUB_*` variables, by default. The runner does not allow it, unless we patch libc